web3/utils/module_testing/eth_module.py

```diff
@@ -159,6 +159,11 @@ def test_eth_getCode_invalid_address(self, web3, math_contract):
         with pytest.raises(InvalidAddress):
             web3.eth.getCode(math_contract.address.lower())
 
+    def test_eth_getCode_with_block_identifier(self, web3, emitter_contract):
+        code = web3.eth.getCode(emitter_contract.address, block_identifier=web3.eth.blockNumber)
+        assert is_string(code)
+        assert len(code) > 2
```

Comment:
These tests read as if the code is a hex string, but it should actually be a `HexBytes` (because of https://github.com/ethereum/web3.py/blob/master/web3/middleware/pythonic.py#L307 ). It passes anyway, because `is_string` accepts `bytes` and the code is longer than two bytes. But I think it could be confusing.

Some alternative options:
1. assert what the code is equal to exactly: `assert code == Web3.toBytes(hexstr='0x`...
2. assert that the code is bytes, and is exactly the expected length:
```py
assert isinstance(code, bytes)
assert len(code) == ...
```

Reply:
@carver I had the same thought in mind when I started writing this, but then I had a look at the surrounding code and found:
https://github.com/ethereum/web3.py/blob/910443d8a17d73841341309f8497893a96ddb3b6/web3/utils/module_testing/eth_module.py#L153-L156
But Looks like that test was written before `Hexbytes` was introduced. I'll change it at both places.

Reply:
> assert code == Web3.toBytes(hexstr='0x')
assert len(code) == ...

I think from the above code you mean to compare deployed code and bytecode runtime. The two need not be the same. I can say this from my previous work on [compare_bytecode](https://github.com/ethereum/populus/pull/471/files#diff-8fa659c73d2daba076463f01c4b4d3e9R170). However, you would know better what is going on internally. The check above won't work.
The lengths don't match too. 
```
(Pdb) len(add_0x_prefix(EMITTER_BYTECODE))
2434
(Pdb) len(code)
2398
```
Or do you mean to say that I should hardcode the length check to 2398 in the test? 

I have changed the `is_string` check to `isinstance(code, HexBytes)`. 
But I kept the length check as is for reasons explained below.  `len > 2` ensures that the bytecode is not empty (`0x`) .

I added a commit to speed up the process if you agree. If you have anything else in mind then I'll revert the last commit.

Reply:
> len > 2 ensures that the bytecode is not empty (0x).

The `0x` is not counted in the length of a `HexBytes` object. If we just want to check non-empty, which is fine, then let's just test `len(code) > 0`

Reply:
Ah, I was fooled by `HexBytes`'s `__repr__`. I didn't actually check the length.

Reply:
@carver Done. squashed all commits.